File: test/single-language/java/gradle/spring-boot/success-json-api/src/main/java/gold/debug/windowstolinux/acceptance/complex/model/Summary.java
package gold.debug.windowstolinux.acceptance.complex.model;

import java.util.List;
import java.util.stream.Collectors;

public record Summary(String status, List<Integer> items, int total) {
    public Summary(List<Integer> items) {
        this("ok", List.copyOf(items), items.stream().mapToInt(Integer::intValue).sum());
    }

    public String toJson() {
        return "{\"status\":\"ok\",\"items\":[" + items.stream().map(String::valueOf).collect(Collectors.joining(","))
                + "],\"total\":" + total + "}";
    }
}

File: test/single-language/kotlin/kotlinc/http-service/success-runtime-config/src/acceptance/model/Summary.kt
package acceptance.model
data class Summary(val items: List<Int>) {
    val status: String = "ok"
    val total: Int = items.sum()

    fun toJson(): String =
        "{\"status\":\"ok\",\"items\":[${items.joinToString(",")}],\"total\":$total}"
}
